### ISSUE TITLE
Error with CFEP 15 unable to pickup `python_min` from variant file

### ISSUE THREAD

**Issue by jakirkham** [prefix-dev/rattler-build#1248]
It looks like `rattler-build` doesn't pick up keys from variant files when determining how to fill out variables like `python_min`

For example take [this recipe]( https://github.com/conda-forge/fastapi-mail-feedstock/blob/9134ce35edfafad6783b17cfb6354614090c4569/recipe/recipe.yaml ) using [this variant file]( https://github.com/conda-forge/fastapi-mail-feedstock/blob/9134ce35edfafad6783b17cfb6354614090c4569/.ci_support/linux_64_.yaml ) (included below in details)

<details>
<summary>
<code>recipe/recipe.yaml</code>
</summary>

```yaml
context:
  version: 1.4.2

package:
  name: fastapi-mail
  version: ${{ version }}

source:
  url: https://pypi.org/packages/source/f/fastapi-mail/fastapi_mail-${{ version }}.tar.gz
  sha256: 04bde1005c624f42dfc0a9c1e313fcc544499fdd6b3531e606c500d80ac2ffcb

build:
  number: 0
  noarch: python
  script: python -m pip install . -vv

requirements:
  host:
    - python ${{ python_min }}
    - poetry-core >=1.2.0
    - pip
  run:
    - python >=${{ python_min }},<4.0
    - jinja2 >=3.0.0,<4.0.0
    - blinker >=1.5.0,<2.0.0
    - starlette >=0.24,<1.0
    - aiosmtplib >=3.0.2,<4.0.0
    - pydantic >=2.10.1,<3.0.0
    - email-validator >=2.2.0,<3.0.0
    - pydantic-settings >=2.6.1,<3.0.0
  run_constraints:
    - httpx >=0.23.0,<0.24.0
    - redis-py >=4.3.0,<5.0.0

tests:
  - python:
      imports:
        - fastapi_mail
      pip_check: true
      python_version: ${{ python_min }}

about:
  summary: Simple lightweight mail library for FastApi
  license: MIT
  license_file: LICENSE
  homepage: https://github.com/sabuhish/fastapi-mail

extra:
  recipe-maintainers:
    - synapticarbors
```

</details>

<details>
<summary>
<code>.ci_support/linux_64_.yaml</code>
</summary>

```yaml
cdt_name:
- conda
channel_sources:
- conda-forge
channel_targets:
- conda-forge main
docker_image:
- quay.io/condaforge/linux-anvil-x86_64:alma9
python_min:
- '3.9'
```

</details>

When built has the following error [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1089290&view=logs&jobId=7b6f2c87-f3a7-5133-8d84-7c03a75d9dfc&j=7b6f2c87-f3a7-5133-8d84-7c03a75d9dfc&t=9eb77fd2-8ddd-5444-8fc0-71cb28dcb736 ) (also [attached log]( https://github.com/user-attachments/files/18069663/fastapi-mail-feedstock_linux_64_20241123.4.txt ) for posterity):

```
rattler-build build --recipe /home/conda/recipe_root -m /home/conda/feedstock_root/.ci_support/linux_64_.yaml --target-platform linux-64 --extra-meta flow_run_id=azure_20241123.4.1 --extra-meta remote_url=*** --extra-meta sha=fb3f2f0cfbeb47b9953fca2ad208f25064a19c23
Error:   × Failed to parse recipe

Error:   × Parsing: failed to parse match spec: unable to parse version spec: >=,<4.0
    ╭─[23:7]
 22 │   run:
 23 │     - python >=${{ python_min }},<4.0
    ·       ───────────────┬───────────────
    ·                      ╰── error parsing `python >=,<4.0` as a match spec
 24 │     - jinja2 >=3.0.0,<4.0.0
    ╰────
```

**Comment by jakirkham** [comment]
The workaround is simply to respecify this variable in `context`

For example...

```yaml
# filename: recipe/recipe.yaml
context:
  ...
  python_min: "3.9"
...
```

**Comment by jakirkham** [comment]
Also as an interesting side note, it looks like there are cases where re-rendering doesn't work as expected either

xref: https://github.com/conda-forge/numba-cuda-feedstock/pull/8#issuecomment-2529920488

**Comment by wolfv** [comment]
Hi @jakirkham - fixed in #1250 and added a test as well :)

**Comment by jakirkham** [comment]
Thanks Wolf! 🙏
